### PR TITLE
Fix I/O issues of subset collections with vector members

### PR DESCRIPTION
**Pull request by tmadlener**
BEGINRELEASENOTES
- Make sure to only access vector member buffers of collections of datatypes with `VectorMembers` if they actually exist. This is necessary to make subset collections of such datatypes work in I/O. Fixes [#462](https://github.com/AIDASoft/podio/issues/462)
- Add a test that reproduces the original issue and is fixed by this.

ENDRELEASENOTES

### CODE REVIEW

File: python/podio/test_Frame.py [hunk 1]
```diff
@@ -16,7 +16,10 @@
     'emptyCollection', 'emptySubsetColl'
     }
 # The expected collections from the extension (only present in the other_events category)
-EXPECTED_EXTENSION_COLL_NAMES = {"extension_Contained", "extension_ExternalComponent", "extension_ExternalRelation"}
+EXPECTED_EXTENSION_COLL_NAMES = {
+    "extension_Contained", "extension_ExternalComponent", "extension_ExternalRelation",
```
Comment (hegner) [inline python/podio/test_Frame.py:20]: can you fix the indentation / formatting?

File: tests/read_frame.h [hunk 2]
```diff
@@ -60,6 +61,14 @@ void processExtensions(const podio::Frame& event, int iEvent, podio::version::Ve
   ASSERT(structs[2].y == 2 * iEvent, "struct value not as expected");
 }
 
+void checkVecMemSubsetColl(const podio::Frame& event) {
+  const auto& subsetColl = event.get<ExampleWithVectorMemberCollection>("VectorMemberSubsetColl");
+  const auto& origColl = event.get<ExampleWithVectorMemberCollection>("WithVectorMember");
+  ASSERT(subsetColl.isSubsetCollection(), "subset collection not read back as a sbuset collection");
```
Comment (hegner) [inline tests/read_frame.h:67]: simple typo in mediate ("sbuset")

**Review by hegner — CHANGES_REQUESTED** [review]
Logic and new tests seem fine to me. Just some nitpicking about formatting and typos. See comments.

**Comment by tmadlener** [comment]
Now also fixing the second issue reported in https://github.com/AIDASoft/podio/issues/462#issuecomment-1649866017

Unless we want to have that in another PR?

File: python/podio/test_Frame.py (hunk 1 above)
Comment (hegner) [reply python/podio/test_Frame.py:22]: Sorry, my bad. Seems my phone display was playing me tricks.

**Pull request merged**